spark_log_parser/loaders/__init__.py

```diff
@@ -283,13 +283,17 @@ def extract(self, filepath: Path, file_stream: BufferedIOBase = None) -> Iterato
             # If the Path is not a directory and has no suffix, it will match []
             case [".json"] | [".log"] | []:
                 logger.info(f"Yielding raw file: {filepath}")
-                if file_stream:
-                    wrapped = FileChunkStreamWrapper(file_stream)
-                    yield filepath, self.read_file_stream(wrapped)
+                match file_stream:
+                    case None:
+                        with open(filepath, "rb") as file:
+                            wrapped = FileChunkStreamWrapper(file)
+                            yield filepath, self.read_file_stream(wrapped)
 
-                else:
-                    with open(filepath, "rb") as file:
-                        wrapped = FileChunkStreamWrapper(file)
+                    case FileChunkStreamWrapper():
```

Comment:
I wanted to avoid double-wrapping in the event that we already have a wrapped stream, and refactoring this to be a `match` for that seemed nicer to read vs. a bunch of if/else statements